Saros/I plugin causes assertion error when closing and then re-opening project

When closing the current project and re-opening it (or another project), the Saros plugin causes an assertion error trying to use the old project object. (The full stacktrace is given below.) This most likely happens because the project object is held in the saros plugin context.

The project object is put into the plugin context through the [IntellijProjectLifecycle](https://github.com/saros-project/saros/blob/master/de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/IntellijProjectLifecycle.java), which in turn calls the [SarosIntellijContextFactory](https://github.com/saros-project/saros/blob/master/de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/context/SarosIntellijContextFactory.java#L105). The lifecycle is instantiated by the [SarosComponent](https://github.com/saros-project/saros/blob/master/de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/SarosComponent.java#L50), which hooks into IntelliJ through the [plugin.xml](https://github.com/saros-project/saros/blob/master/de.fu_berlin.inf.dpp.intellij/META-INF/plugin.xml#L31)

The question now is why this context is not cleaned up between the project changes. Saros/I should be a project level component, meaning a new instance of the plugin should be created once the new project is opened, then also containing the correct project object.

#### Stacktrace
```Java
java.lang.AssertionError: Already disposed: Project (Disposed) HostTestProject
	at com.intellij.openapi.components.impl.ComponentManagerImpl.lambda$throwAlreadyDisposed$1(ComponentManagerImpl.java:248)
	at com.intellij.openapi.application.ReadAction.lambda$run$1(ReadAction.java:53)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:955)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:57)
	at com.intellij.openapi.application.ReadAction.run(ReadAction.java:53)
	at com.intellij.openapi.components.impl.ComponentManagerImpl.throwAlreadyDisposed(ComponentManagerImpl.java:246)
	at com.intellij.openapi.components.impl.ComponentManagerImpl.getPicoContainer(ComponentManagerImpl.java:239)
	at com.intellij.openapi.components.impl.ComponentManagerImpl.getPicoContainer(ComponentManagerImpl.java:45)
	at com.intellij.openapi.components.ServiceManager.doGetService(ServiceManager.java:42)
	at com.intellij.openapi.components.ServiceManager.getService(ServiceManager.java:30)
	at com.intellij.openapi.roots.ProjectFileIndex.getInstance(ProjectFileIndex.java:44)
	at com.intellij.openapi.roots.ProjectFileIndex$SERVICE.getInstance(ProjectFileIndex.java:38)
	at com.intellij.openapi.module.ModuleUtilCore.findModuleForFile(ModuleUtilCore.java:71)
	at de.fu_berlin.inf.dpp.intellij.filesystem.VirtualFileConverter.convertToResource(VirtualFileConverter.java:63)
	at de.fu_berlin.inf.dpp.intellij.filesystem.VirtualFileConverter.convertToSPath(VirtualFileConverter.java:46)
	at de.fu_berlin.inf.dpp.intellij.editor.LocalEditorHandler.activateEditor(LocalEditorHandler.java:221)
	at de.fu_berlin.inf.dpp.intellij.editor.LocalEditorStatusChangeHandler.generateEditorActivatedActivity(LocalEditorStatusChangeHandler.java:150)
	at de.fu_berlin.inf.dpp.intellij.editor.LocalEditorStatusChangeHandler.access$300(LocalEditorStatusChangeHandler.java:18)
	at de.fu_berlin.inf.dpp.intellij.editor.LocalEditorStatusChangeHandler$1.selectionChanged(LocalEditorStatusChangeHandler.java:44)
	at sun.reflect.GeneratedMethodAccessor34.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.util.messages.impl.MessageBusConnectionImpl.deliverMessage(MessageBusConnectionImpl.java:117)
	at com.intellij.util.messages.impl.MessageBusImpl.doPumpMessages(MessageBusImpl.java:426)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpWaitingBuses(MessageBusImpl.java:387)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpMessages(MessageBusImpl.java:376)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpMessages(MessageBusImpl.java:364)
	at com.intellij.util.messages.impl.MessageBusImpl.sendMessage(MessageBusImpl.java:357)
	at com.intellij.util.messages.impl.MessageBusImpl.access$200(MessageBusImpl.java:43)
	at com.intellij.util.messages.impl.MessageBusImpl$2.invoke(MessageBusImpl.java:208)
	at com.sun.proxy.$Proxy101.selectionChanged(Unknown Source)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.lambda$fireSelectionChanged$16(FileEditorManagerImpl.java:1584)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl$5$1.run(FileEditorManagerImpl.java:1033)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.lambda$doWhenFocusSettlesDown$2(FocusManagerImpl.java:199)
	at com.intellij.util.ui.UIUtil.invokeLaterIfNeeded(UIUtil.java:3094)
	at com.intellij.ide.IdeEventQueue.ifFocusEventsInTheQueue(IdeEventQueue.java:183)
	at com.intellij.ide.IdeEventQueue.executeWhenAllFocusEventsLeftTheQueue(IdeEventQueue.java:132)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.doWhenFocusSettlesDown(FocusManagerImpl.java:190)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.doWhenFocusSettlesDown(FocusManagerImpl.java:196)
	at com.intellij.openapi.wm.impl.IdeFocusManagerImpl.doWhenFocusSettlesDown(IdeFocusManagerImpl.java:63)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl$5.run(FileEditorManagerImpl.java:1030)
	at com.intellij.openapi.util.BusyObject$Impl$Simple.execute(BusyObject.java:108)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.notifyPublisher(FileEditorManagerImpl.java:1026)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.fireSelectionChanged(FileEditorManagerImpl.java:1584)
	at com.intellij.openapi.fileEditor.impl.EditorsSplitters.lambda$setCurrentWindow$2(EditorsSplitters.java:599)
	at com.intellij.openapi.fileEditor.impl.EditorsSplitters.setCurrentWindow(EditorsSplitters.java:608)
	at com.intellij.openapi.fileEditor.impl.EditorWindow.setEditor(EditorWindow.java:722)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.lambda$openFileImpl4$9(FileEditorManagerImpl.java:913)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:315)
	at com.intellij.openapi.application.impl.LaterInvocator$1.run(LaterInvocator.java:156)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.doRun(LaterInvocator.java:447)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.runNextEvent(LaterInvocator.java:431)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:415)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:781)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:722)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:382)
	at com.intellij.ide.IdeEventQueue.pumpEventsForHierarchy(IdeEventQueue.java:864)
	at com.intellij.openapi.progress.util.ProgressWindow.startBlocking(ProgressWindow.java:205)
	at com.intellij.openapi.progress.util.ProgressWindow.startBlocking(ProgressWindow.java:191)
	at com.intellij.openapi.application.impl.ApplicationImpl.runProcessWithProgressSynchronously(ApplicationImpl.java:588)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcessWithProgressSynchronously(CoreProgressManager.java:446)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.runProcessWithProgressSynchronously(ProgressManagerImpl.java:109)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcessWithProgressSynchronously(CoreProgressManager.java:250)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcessWithProgressSynchronously(CoreProgressManager.java:203)
	at com.intellij.openapi.project.impl.ProjectManagerImpl.loadProjectUnderProgress(ProjectManagerImpl.java:439)
	at com.intellij.openapi.project.impl.ProjectManagerImpl.openProject(ProjectManagerImpl.java:414)
	at com.intellij.platform.PlatformProjectOpenProcessor.doOpenProject(PlatformProjectOpenProcessor.java:245)
	at com.intellij.ide.RecentProjectsManagerBase.doOpenProject(RecentProjectsManagerBase.java:579)
	at com.intellij.ide.ReopenProjectAction.actionPerformed(ReopenProjectAction.java:72)
	at com.intellij.openapi.wm.impl.welcomeScreen.RecentProjectPanel$3.onClick(RecentProjectPanel.java:156)
	at com.intellij.ui.ClickListener$1.mouseReleased(ClickListener.java:73)
	at java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:290)
	at java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:289)
	at java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:289)
	at java.awt.Component.processMouseEvent(Component.java:6539)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
	at java.awt.Component.processEvent(Component.java:6304)
	at java.awt.Container.processEvent(Container.java:2239)
	at java.awt.Component.dispatchEventImpl(Component.java:4889)
	at java.awt.Container.dispatchEventImpl(Container.java:2297)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4904)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4535)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4476)
	at java.awt.Container.dispatchEventImpl(Container.java:2283)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:760)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:84)
	at java.awt.EventQueue$4.run(EventQueue.java:733)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:730)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:781)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:718)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:382)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

--- a/intellij/resources/META-INF/plugin.xml
+++ b/intellij/resources/META-INF/plugin.xml
@@ -34,15 +34,12 @@
     -->
 
     <application-components>
-        <!-- Add your application components here -->
-    </application-components>
-
-    <project-components>
         <component>
-            <implementation-class>saros.intellij.SarosComponent
+            <implementation-class>
+                saros.intellij.SarosComponent
             </implementation-class>
         </component>
-    </project-components>
+    </application-components>
 
     <actions>
         <group keep-content="true" compact="false" popup="true"

--- a/intellij/src/saros/core/util/IntellijCollaborationUtilsImpl.java
+++ b/intellij/src/saros/core/util/IntellijCollaborationUtilsImpl.java
@@ -1,8 +1,13 @@
 package saros.core.util;
 
+import com.intellij.openapi.project.Project;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 import saros.core.ui.util.CollaborationUtils;
 import saros.filesystem.IResource;
+import saros.intellij.context.SharedIDEContext;
+import saros.intellij.filesystem.IntelliJProjectImpl;
 import saros.intellij.ui.util.UIProjectUtils;
 import saros.net.xmpp.JID;
 import saros.ui.util.ICollaborationUtils;
@@ -22,6 +27,21 @@ public class IntellijCollaborationUtilsImpl implements ICollaborationUtils {
 
   @Override
   public void startSession(List<IResource> resources, List<JID> contacts) {
+    Set<Project> sharedProjects = new HashSet<>();
+
+    for (IResource resource : resources) {
+      sharedProjects.add(
+          resource.getProject().adaptTo(IntelliJProjectImpl.class).getModule().getProject());
+    }
+
+    // Workaround while we only allow a single module (-> single project) to be shared
+    if (sharedProjects.size() != 1) {
+      throw new IllegalStateException(
+          "Saros/I currently only supports sharing of a single project. Found projects: "
+              + sharedProjects);
+    }
+    SharedIDEContext.preregisterProject(sharedProjects.toArray(new Project[1])[0]);
+
     CollaborationUtils.startSession(resources, contacts);
   }
 

--- a/intellij/src/saros/intellij/IntellijProjectLifecycle.java
+++ b/intellij/src/saros/intellij/IntellijProjectLifecycle.java
@@ -1,6 +1,5 @@
 package saros.intellij;
 
-import com.intellij.openapi.project.Project;
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.List;
@@ -26,28 +25,24 @@ public class IntellijProjectLifecycle extends AbstractContextLifecycle {
   private static IntellijProjectLifecycle instance;
 
   /**
-   * Creates a new IntellijProjectLifecycle singleton instance from a project.
+   * Returns the current intellij project lifecycle instance. The returned instance is singleton. If
+   * no instance is present, a new instance is instantiated.
    *
-   * @param project
-   * @return
+   * @return the current intellij project lifecycle instance
    */
-  public static synchronized IntellijProjectLifecycle getInstance(Project project) {
-    instance = new IntellijProjectLifecycle(project);
+  public static synchronized IntellijProjectLifecycle getInstance() {
+    if (instance == null) {
+      instance = new IntellijProjectLifecycle();
+    }
 
     return instance;
-  }
-
-  private Project project;
-
-  private IntellijProjectLifecycle(Project project) {
-    this.project = project;
   }
 
   @Override
   protected Collection<IContextFactory> additionalContextFactories() {
     List<IContextFactory> nonCoreFactories = new ArrayList<>();
 
-    nonCoreFactories.add(new SarosIntellijContextFactory(project));
+    nonCoreFactories.add(new SarosIntellijContextFactory());
 
     if (SarosComponent.isSwtBrowserEnabled()) {
       SwtLibLoader.loadSwtLib();

--- a/intellij/src/saros/intellij/SarosComponent.java
+++ b/intellij/src/saros/intellij/SarosComponent.java
@@ -3,19 +3,17 @@ package saros.intellij;
 import com.intellij.openapi.actionSystem.KeyboardShortcut;
 import com.intellij.openapi.keymap.Keymap;
 import com.intellij.openapi.keymap.KeymapManager;
-import com.intellij.openapi.project.Project;
 import java.awt.event.KeyEvent;
 import java.io.InputStream;
 import javax.swing.KeyStroke;
 import org.apache.log4j.PropertyConfigurator;
 import org.apache.log4j.helpers.LogLog;
-import org.jetbrains.annotations.NotNull;
 
 /**
- * Component that is initalized when a project is loaded. It initializes the logging, shortcuts and
+ * Component that is initialized when a project is loaded. It initializes the logging, shortcuts and
  * the {@link IntellijProjectLifecycle} singleton.
  */
-public class SarosComponent implements com.intellij.openapi.components.ProjectComponent {
+public class SarosComponent {
 
   /**
    * This is the plugin ID that identifies the saros plugin in the IDEA ecosystem. It is set in
@@ -23,7 +21,7 @@ public class SarosComponent implements com.intellij.openapi.components.ProjectCo
    */
   public static final String PLUGIN_ID = "saros";
 
-  public SarosComponent(final Project project) {
+  public SarosComponent() {
     loadLoggers();
 
     Keymap keymap = KeymapManager.getInstance().getActiveKeymap();
@@ -50,7 +48,7 @@ public class SarosComponent implements com.intellij.openapi.components.ProjectCo
       LogLog.error("could not load saros property file 'saros.properties'", e);
     }
 
-    IntellijProjectLifecycle.getInstance(project).start();
+    IntellijProjectLifecycle.getInstance().start();
   }
 
   public static boolean isSwtBrowserEnabled() {
@@ -72,31 +70,5 @@ public class SarosComponent implements com.intellij.openapi.components.ProjectCo
     } finally {
       Thread.currentThread().setContextClassLoader(contextClassLoader);
     }
-  }
-
-  @Override
-  public void initComponent() {
-    // NOP
-  }
-
-  @Override
-  public void disposeComponent() {
-    // NOP
-  }
-
-  @NotNull
-  @Override
-  public String getComponentName() {
-    return "Saros";
-  }
-
-  @Override
-  public void projectOpened() {
-    // TODO: Update project
-  }
-
-  @Override
-  public void projectClosed() {
-    // TODO: Update project
   }
 }

--- a/intellij/src/saros/intellij/context/SarosIntellijContextFactory.java
+++ b/intellij/src/saros/intellij/context/SarosIntellijContextFactory.java
@@ -1,6 +1,5 @@
 package saros.intellij.context;
 
-import com.intellij.openapi.project.Project;
 import java.util.Arrays;
 import saros.communication.connection.IProxyResolver;
 import saros.communication.connection.NullProxyResolver;
@@ -80,19 +79,12 @@ public class SarosIntellijContextFactory extends AbstractContextFactory {
     };
   }
 
-  private Project project;
-
-  public SarosIntellijContextFactory(Project project) {
-    this.project = project;
-  }
-
   @Override
   public void createComponents(MutablePicoContainer container) {
 
     // Saros Core PathIntl Support
     container.addComponent(IPathFactory.class, new PathFactory());
 
-    container.addComponent(Project.class, project);
     container.addComponent(IWorkspace.class, IntelliJWorkspaceImpl.class);
 
     for (Component component : Arrays.asList(getContextComponents())) {

--- a/intellij/src/saros/intellij/context/SharedIDEContext.java
+++ b/intellij/src/saros/intellij/context/SharedIDEContext.java
@@ -36,17 +36,13 @@ public class SharedIDEContext implements Disposable {
 
   public SharedIDEContext(
       ApplicationEventHandlersFactory applicationEventHandlersFactory,
-      ProjectEventHandlersFactory projectEventHandlersFactory,
-      Project project) {
+      ProjectEventHandlersFactory projectEventHandlersFactory) {
 
     this.applicationEventHandlersFactory = applicationEventHandlersFactory;
     this.projectEventHandlersFactory = projectEventHandlersFactory;
 
     if (preregisteredProject != null) {
       startProjectComponents(preregisteredProject);
-    } else {
-      // TODO remove this workaround once the shared project object is correctly initialized
-      startProjectComponents(project);
     }
   }
 

--- a/intellij/src/saros/intellij/ui/menu/ShareWithUserAction.java
+++ b/intellij/src/saros/intellij/ui/menu/ShareWithUserAction.java
@@ -13,6 +13,7 @@ import java.util.List;
 import org.apache.log4j.Logger;
 import saros.core.ui.util.CollaborationUtils;
 import saros.filesystem.IResource;
+import saros.intellij.context.SharedIDEContext;
 import saros.intellij.filesystem.IntelliJProjectImpl;
 import saros.intellij.ui.Messages;
 import saros.intellij.ui.util.IconManager;
@@ -46,17 +47,25 @@ public class ShareWithUserAction extends AnAction {
 
   @Override
   public void actionPerformed(AnActionEvent e) {
-    VirtualFile virtFile = e.getData(CommonDataKeys.VIRTUAL_FILE);
-    if (virtFile == null) {
-      return;
+    Project project = e.getProject();
+    if (project == null) {
+      throw new IllegalStateException(
+          "Unable to start session - could not determine project for highlighted resource.");
+    }
+
+    VirtualFile virtualFile = e.getData(CommonDataKeys.VIRTUAL_FILE);
+    if (virtualFile == null) {
+      throw new IllegalStateException(
+          "Unable to start session - could not determine virtual file for highlighted resource.");
     }
 
     // We allow only completely shared projects, so no need to check
     // for partially shared ones.
-    List<IResource> resources = Arrays.asList(getModuleFromVirtFile(virtFile, e.getProject()));
+    List<IResource> resources = Arrays.asList(getModuleFromVirtFile(virtualFile, e.getProject()));
 
     List<JID> contacts = Arrays.asList(userJID);
 
+    SharedIDEContext.preregisterProject(project);
     CollaborationUtils.startSession(resources, contacts);
   }
 

--- a/intellij/src/saros/intellij/ui/wizards/AddProjectToSessionWizard.java
+++ b/intellij/src/saros/intellij/ui/wizards/AddProjectToSessionWizard.java
@@ -32,6 +32,7 @@ import org.jetbrains.annotations.Nullable;
 import saros.SarosPluginContext;
 import saros.filesystem.IChecksumCache;
 import saros.filesystem.IProject;
+import saros.intellij.context.SharedIDEContext;
 import saros.intellij.editor.DocumentAPI;
 import saros.intellij.filesystem.Filesystem;
 import saros.intellij.filesystem.IntelliJProjectImpl;
@@ -75,7 +76,6 @@ import saros.util.ThreadUtils;
  */
 
 //  FIXME: Add facility for more than one project.
-
 public class AddProjectToSessionWizard extends Wizard {
   private static final Logger LOG = Logger.getLogger(AddProjectToSessionWizard.class);
 
@@ -137,6 +137,8 @@ public class AddProjectToSessionWizard extends Wizard {
           }
 
           Project project = moduleSelectionResult.getProject();
+
+          sessionManager.getSession().getComponent(SharedIDEContext.class).setProject(project);
 
           switch (moduleSelectionResult.getLocalRepresentationOption()) {
             case CREATE_NEW_MODULE:

--- a/intellij/test/junit/saros/intellij/context/AbstractContextTest.java
+++ b/intellij/test/junit/saros/intellij/context/AbstractContextTest.java
@@ -4,7 +4,6 @@ import static saros.intellij.test.IntellijMocker.mockStaticGetInstance;
 
 import com.intellij.ide.util.PropertiesComponent;
 import com.intellij.openapi.module.ModuleTypeManager;
-import com.intellij.openapi.project.Project;
 import org.easymock.EasyMock;
 import org.junit.Before;
 import org.junit.runner.RunWith;
@@ -19,7 +18,6 @@ import saros.test.mocks.ContextMocker;
 public class AbstractContextTest {
 
   MutablePicoContainer container;
-  Project project;
 
   @Before
   public void setup() {
@@ -28,10 +26,6 @@ public class AbstractContextTest {
     // mock IntelliJ dependencies
     mockStaticGetInstance(PropertiesComponent.class, null);
     mockStaticGetInstance(ModuleTypeManager.class, null);
-
-    project = EasyMock.createNiceMock(Project.class);
-
-    EasyMock.replay(project);
 
     // mock IntelliJ dependent calls to get current IDE and plugin version
     PowerMock.mockStaticPartial(

--- a/intellij/test/junit/saros/intellij/context/SarosIntellijContextFactoryTest.java
+++ b/intellij/test/junit/saros/intellij/context/SarosIntellijContextFactoryTest.java
@@ -24,7 +24,7 @@ public class SarosIntellijContextFactoryTest extends AbstractContextTest {
 
   @Test
   public void testCreateComponents() {
-    IContextFactory factory = new SarosIntellijContextFactory(project);
+    IContextFactory factory = new SarosIntellijContextFactory();
 
     factory.createComponents(container);
     container.start();

--- a/intellij/test/junit/saros/intellij/context/SarosIntellijContextTest.java
+++ b/intellij/test/junit/saros/intellij/context/SarosIntellijContextTest.java
@@ -27,7 +27,7 @@ public class SarosIntellijContextTest extends AbstractContextTest {
   public void createComponentsWithoutSWT() {
     List<IContextFactory> factories = new ArrayList<>();
 
-    factories.add(new SarosIntellijContextFactory(project));
+    factories.add(new SarosIntellijContextFactory());
     factories.add(new CoreContextFactory());
 
     for (IContextFactory factory : factories) {
@@ -43,7 +43,7 @@ public class SarosIntellijContextTest extends AbstractContextTest {
   public void createComponentsWithSWT() {
     List<IContextFactory> factories = new ArrayList<>();
 
-    factories.add(new SarosIntellijContextFactory(project));
+    factories.add(new SarosIntellijContextFactory());
     factories.add(new CoreContextFactory());
     factories.add(new HTMLUIContextFactory());
 

